src/server/http/index.js

```diff
@@ -75,12 +75,21 @@ export default async function (kbnServer, server, config) {
   server.ext('onPreResponse', function (req, reply) {
     const response = req.response;
 
+    const customResponseHeaders = config.get('server.customResponseHeaders');
+
     if (response.isBoom) {
-      response.output.headers['kbn-name'] = kbnServer.name;
-      response.output.headers['kbn-version'] = kbnServer.version;
+      response.output.headers = {
+        ...response.output.headers,
+        ...customResponseHeaders,
```

Comment:
In this branch the `kbn-name` and `kbn-version` headers overwrite the same keys in `customResponseHeaders`, but in the branch below the opposite is true.

Perhaps you could merge `customResponseHeaders` with `kbn-name` and `kbn-version` outside of this condition, then apply the update as necessary within the branches of the condition?